[2024-06-27T14:58:12.847+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-27T14:58:13.080+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: TEST_SPARK.extract.extract_shippers manual__2024-06-27T14:52:46.413909+00:00 [queued]>
[2024-06-27T14:58:13.224+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: TEST_SPARK.extract.extract_shippers manual__2024-06-27T14:52:46.413909+00:00 [queued]>
[2024-06-27T14:58:13.230+0000] {taskinstance.py:2306} INFO - Starting attempt 2 of 2
[2024-06-27T14:58:13.401+0000] {taskinstance.py:2330} INFO - Executing <Task(_PythonDecoratedOperator): extract.extract_shippers> on 2024-06-27 14:52:46.413909+00:00
[2024-06-27T14:58:13.482+0000] {logging_mixin.py:188} WARNING - /home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=4062) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-06-27T14:58:13.497+0000] {standard_task_runner.py:90} INFO - Running: ['airflow', 'tasks', 'run', 'TEST_SPARK', 'extract.extract_shippers', 'manual__2024-06-27T14:52:46.413909+00:00', '--job-id', '444', '--raw', '--subdir', 'DAGS_FOLDER/elt-spark.py', '--cfg-path', '/tmp/tmp7gnnwm2o']
[2024-06-27T14:58:13.511+0000] {standard_task_runner.py:91} INFO - Job 444: Subtask extract.extract_shippers
[2024-06-27T14:58:13.509+0000] {standard_task_runner.py:63} INFO - Started process 4156 to run task
[2024-06-27T14:58:13.675+0000] {logging_mixin.py:188} WARNING - /home/airflow/.local/lib/python3.12/site-packages/airflow/settings.py:195 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2024-06-27T14:58:14.201+0000] {task_command.py:426} INFO - Running <TaskInstance: TEST_SPARK.extract.extract_shippers manual__2024-06-27T14:52:46.413909+00:00 [running]> on host airflow-scheduler
[2024-06-27T14:58:15.084+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Data-Ninja' AIRFLOW_CTX_DAG_ID='TEST_SPARK' AIRFLOW_CTX_TASK_ID='extract.extract_shippers' AIRFLOW_CTX_EXECUTION_DATE='2024-06-27T14:52:46.413909+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-06-27T14:52:46.413909+00:00'
[2024-06-27T14:58:15.099+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-27T14:58:31.121+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-27T14:58:31.124+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/decorators/base.py", line 265, in execute
    return_value = super().execute(context)
                   ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/elt-spark.py", line 41, in extract_task
    return extract_func(
           ^^^^^^^^^^^^^
  File "/opt/airflow/dags/spark/extractors.py", line 25, in extract_func
    .getOrCreate()
     ^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/sql/session.py", line 497, in getOrCreate
    sc = SparkContext.getOrCreate(sparkConf)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/context.py", line 515, in getOrCreate
    SparkContext(conf=conf or SparkConf())
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/context.py", line 201, in __init__
    SparkContext._ensure_initialized(self, gateway=gateway, conf=conf)
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/context.py", line 436, in _ensure_initialized
    SparkContext._gateway = gateway or launch_gateway(conf)
                                       ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/java_gateway.py", line 107, in launch_gateway
    raise PySparkRuntimeError(
pyspark.errors.exceptions.base.PySparkRuntimeError: [JAVA_GATEWAY_EXITED] Java gateway process exited before sending its port number.
[2024-06-27T14:58:31.205+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=TEST_SPARK, task_id=extract.extract_shippers, run_id=manual__2024-06-27T14:52:46.413909+00:00, execution_date=20240627T145246, start_date=20240627T145813, end_date=20240627T145831
[2024-06-27T14:58:31.288+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 444 for task extract.extract_shippers ([JAVA_GATEWAY_EXITED] Java gateway process exited before sending its port number.; 4156)
[2024-06-27T14:58:31.340+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-27T14:58:31.563+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-06-27T14:58:31.604+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
